static/app/components/events/groupingInfo/index.tsx

```diff
@@ -172,49 +113,56 @@ class GroupingInfo extends DeprecatedAsyncComponent<Props, State> {
     return (
       <Fragment>
         <ConfigHeader>
-          <div>{showGroupingConfig && this.renderGroupConfigSelect()}</div>
+          <div>{showGroupingConfig && renderGroupConfigSelect()}</div>
           <FeatureFeedback
             featureName="grouping"
             feedbackTypes={groupingFeedbackTypes}
             buttonProps={{size: 'sm'}}
           />
         </ConfigHeader>
-
-        {loading ? (
-          <LoadingIndicator />
-        ) : (
-          variants.map((variant, index) => (
-            <Fragment key={variant.key}>
-              <GroupVariant
-                event={event}
-                variant={variant}
-                showGroupingConfig={showGroupingConfig}
-              />
-              {index < variants.length - 1 && <VariantDivider />}
-            </Fragment>
-          ))
-        )}
+        {isError ? <LoadingError message={t('Failed to fetch grouping info.')} /> : null}
+        {isLoading && !hasPerformanceGrouping ? <LoadingIndicator /> : null}
+        {hasPerformanceGrouping || isSuccess
+          ? variants.map((variant, index) => (
```

Comment:
Can we surround this `map` with useMemo?

Reply:
this component shouldn't rerender outside of loading? what would useMemo give us

Reply:
With the current change, whenever the component re-renders, the `variants.map` gets executed. If we put it inside a `useMemo` and give `hasPerformanceGrouping, isSuccess` as the dependencies, without these dependencies it will never get re-calculated.

To summarize, I'm trying to reduce the unnecessary re-calculations on re-renders.

Reply:
Are you sure this is necessary? I'd rather keep it as is for simplicity, there will only ever be 1-3 elements in the array

Reply:
its an array of like 10 things and probably won't rerender once its loaded, trying to avoid useMemo and useCallback being slammed into every place

Reply:
I understand. It is not necessary, and non-blocking. But let's not forget that if there is 50-100 (whatever the number is) components in the vdom that follows this approach, it will result in perf. problems. 

It will be a "death by thousand cuts".

I read this article by Kent C. Dodds every once in a while, and it helped me a lot. https://kentcdodds.com/blog/usememo-and-usecallback